packages/perseus/src/widgets/interactive-graphs/math/geometry.test.ts

```diff
@@ -0,0 +1,178 @@
+import {segmentsIntersect} from "./geometry";
+
+import type {Segment} from "./geometry";
+
+describe("segmentsIntersect", () => {
+    it("returns false when segments have zero length", () => {
+        const segment1: Segment = [
+            [0, 0],
+            [0, 0],
+        ];
+        const segment2: Segment = [
+            [1, 1],
+            [1, 1],
+        ];
+        expect(segmentsIntersect(segment1, segment2)).toBe(false);
+    });
+
+    it("returns false when segments are the same", () => {
+        // TODO(benchristel): is this desired behavior?
```

Comment:
I _think_ it is mathematically.  [This](https://www.cuemath.com/geometry/intersection-of-two-lines/#) page says that intersecting lines always meet at a single point (two identical lines would meet at all points on both lines).

It also states that "The intersecting lines can cross each other at any angle. This angle formed is always greater than 0° and less than 180°." Again, identical lines would form an angle of 0°.

That page is basically the first hit for "do identical lines intersect" so it isn't authoritative... but seems to make logical sense. :)